app/models/concerns/client_search.rb

```diff
@@ -52,6 +52,10 @@ def self.text_searcher(text, &block)
         where = metaphone_search(where, :FirstName, text)
         where = metaphone_search(where, :LastName, text)
       end
+
+      # Model may define a function 'injected_search' to add additional clauses
+      where = injected_search(where, text, numeric: numeric, alpha_numeric: alpha_numeric, date: date) if respond_to?(:injected_search)
```

Comment:
I don't know about this, it feels kind of silly. Should it just be called `external_id_search`? Is this what you had in mind @eanders ?

Reply:
Yeah, I think this needs to be more specific, otherwise you can only inject one of these....
what about:
```suggestion
      where = search_by_external_id(where, text, numeric: numeric, alpha_numeric: alpha_numeric, date: date) if RailsDrivers.loaded.include?(:hmis_external_apis)
```

Reply:
Ok. We still need to ask `respond_to?(:search_by_external_id)` because the grdawarehouse client doesnt have the external ID function (yet)

Reply:
Any reason this shouldn't work for both kinds of clients?  Seems like we'll want this same feature on the warehouse side.

Reply:
Yeah I think we will want it, I just didn't do that work in this PR. That will involve making a second ClientExtension for the GrdaWarehouse client, I think.

Reply:
Ok, should basically be copy-paste right?  Maybe a concern?